lib/functions/cli/cli-patch.sh

```diff
@@ -123,4 +108,39 @@ function cli_patch_uboot_run() {
 	LOG_SECTION="patch_uboot_target" do_with_logging patch_uboot_target
 
 	display_alert "Done patching u-boot" "${BRANCH} - ${LINUXFAMILY} - ${BOOTSOURCE}#${BOOTBRANCH}" "cachehit"
+
+	if [[ "${do_push}" == "yes" ]]; then
+		display_alert "Pushing u-boot to Git branch ${target_branch}" "${target_repo_url}" "info"
+		git_ensure_safe_directory "${SRC}/cache/git-bare/u-boot"
+		push_command=(git -C "${SRC}/cache/git-bare/u-boot" push "--force" "--verbose" "${target_repo_url}" "u-boot-${BRANCH}-${BOARD}:${target_branch}")
+		display_alert "Git push command: " "${push_command[*]}" "info"
+		execute_git_push
+	fi
+
+}
+
+function determine_git_push_details() {
+	if [[ -n "${PUSH_TO_GITHUB}" ]]; then
+		PUSH_TO_REPO="git@github.com:${PUSH_TO_GITHUB}.git"
+		display_alert "Will push to GitHub" "${PUSH_TO_REPO}" "info"
+	fi
+
+	if [[ -n "${PUSH_TO_REPO}" ]]; then
+		do_push="yes"
+		declare ymd vendor_lc
+		ymd="$(date +%Y%m%d)"
+		vendor_lc="$(tr '[:upper:]' '[:lower:]' <<< "${VENDOR}" | tr ' ' '_')" # lowercase ${VENDOR} and replace spaces with underscores
+		target_branch="${vendor_lc}-${1}-${ymd}${PUSH_BRANCH_POSTFIX:-""}"
+		target_repo_url="${PUSH_TO_REPO}"
+		display_alert "Will push to Git" "${target_repo_url} branch ${target_branch}" "info"
+	else
+		display_alert "Will NOT push to Git" "use PUSH_TO_GITHUB=org/repo or PUSH_TO_REPO=<url> to push" "info"
+	fi
+}
+
+function execute_git_push() {
+	display_alert "Pushing to ${target_branch}" "${target_repo_url}" "info"
+	# @TODO: do NOT allow shallow trees here, we need the full history to be able to push
+	GIT_SSH_COMMAND="ssh -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" "${push_command[@]}"
```

Comment:
Hmm. This is from a few years ago, I just refactored it into a function.

I read it as "this is gonna run inside a Docker container which never talked to github.com over ssh before and try to push over ssh, so make ssh complacent". The alternative is to let ssh try to interactively ask the user to trust the ssh-host key for github.com (and fail, since the thing is not interactive).

Reply:
There's provisions in the Docker launching for an SSH agent proxy. 
Also provisions for including `.git` (large/slow!) in the Dockerfile so we can do "git archeology" when rewriting patching. Look for

```bash
	declare -g DOCKER_PASS_SSH_AGENT="yes"              # Pass SSH agent to docker
	declare -g DOCKER_PASS_GIT="yes"                    # mount .git dir to docker; for archeology
```

across the codebase.

Reply:
> I assume this doesn't work with the normal GitHub auth?

AFAIK there's no more any "normal" (non-SSH) auth for pushing to GitHub? They deprecated and removed password authentication over https a few years ago if I'm not mistaken...

Reply:
There's gotta be PAT based stuff that works for http.  I can't imagine a world where there's zero options over HTTPS

Reply:
Oh, I guess if you're using the `gh` cli it sets up using a PAT and https; but probably that won't work inside a Docker container unless we somehow pass-through that to the container filesystem.

Either way having the ssh tolerance there shouldn't impact https.

Reply:
Well it's a question of finding what it is that the `gh` cli, or the VS Code stuff, puts the PAT and configs, and hacking into armbian's docker launcher to mount or somehow inject those into the container.

Reply:
Dunno if they changed to use some platform-provided keychain, but it _used to be_ just a token in a file on disk, unencrypted.

Reply:
If you run `gh auth token` it should spit out the token from wherever they store it now, which varies per-platform. On MacOS it's on the Apple Keychain, apparently, which is great & TPM protected. 
Still, Armbian could capture the token in the docker launcher and pass it down in `.gitconfig` or such, for full transparent pushing.
I personally use SSH so that's what I have in this PR, but contributions are welcome.